policy/modules/services/container.if

```diff
@@ -0,0 +1,840 @@
+#
+interface(`container_domtrans_podman',`
+	gen_require(`
+		type podman_t, podman_exec_t;
+	')
+
+	corecmd_search_bin($1)
+	domtrans_pattern($1, podman_exec_t, podman_t)
+')
+
+########################################
+## <summary>
+##	Execute podman in the podman domain,
+##	and allow the specified role the
+##	container domain.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed to transition
+##	</summary>
+## </param>
+## <param name="role">
+##	<summary>
+##	The role to be allowed the container domain.
+##	</summary>
+## </param>
+#
+interface(`container_run_podman',`
+	gen_require(`
+		attribute_role container_roles;
+	')
+
+	container_domtrans_podman($1)
+	roleattribute $2 container_roles;
+')
+
+########################################
+## <summary>
+##	Execute podman in the podman user domain (rootless podman).
+## </summary>
+## <param name="domain">
+## <summary>
+##	Domain allowed to transition.
+## </summary>
+## </param>
+#
+interface(`container_domtrans_podman_user',`
+	gen_require(`
+		type podman_user_t, podman_exec_t;
+	')
+
+	corecmd_search_bin($1)
+	domtrans_pattern($1, podman_exec_t, podman_user_t)
+')
+
+########################################
+## <summary>
+##	Execute podman in the podman user domain,
+##	and allow the specified role the container
+##	domain (rootless podman).
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed to transition
+##	</summary>
+## </param>
+## <param name="role">
+##	<summary>
+##	The role to be allowed the container domain.
+##	</summary>
+## </param>
+#
+interface(`container_run_podman_user',`
+	gen_require(`
+		attribute_role container_roles;
+	')
+
+	container_domtrans_podman_user($1)
+	roleattribute $2 container_roles;
+')
+
+########################################
+## <summary>
+##	Send and receive messages from
+##	container engines over dbus.
+## </summary>
+## <param name="domain">
+## <summary>
+##	Domain allowed access.
+## </summary>
+## </param>
+#
+interface(`container_engine_dbus_chat',`
+	gen_require(`
+		attribute container_engine_domain;
+		class dbus send_msg;
+	')
+
+	allow $1 container_engine_domain:dbus send_msg;
+	allow container_engine_domain $1:dbus send_msg;
+')
+
+########################################
+## <summary>
+##	All of the permissions necessary
+##	for a container engine to manage
+##	container processes.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`container_manage_process',`
+	gen_require(`
+		attribute container_domain;
+	')
+
+	allow $1 container_domain:process { getattr getsched setsched transition signal signull sigkill };
+')
+
+########################################
+## <summary>
+##	Allow the specified domain to
+##	perform a type transition to
+##	container domains.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed to transition.
+##	</summary>
+## </param>
+#
+interface(`container_transition',`
```

Comment:
If there's not a type_transition or setexec how would this work?

Reply:
This interface was added to address this access which was moved from the virt module: https://github.com/SELinuxProject/refpolicy/blob/master/policy/modules/services/virt.te#L866

I can rewrite this to use `domain_transition_pattern()` instead but I'm not really sure what the original executable file was for `virsh_t` to transition to `svirt_lxc_net_t`. I assume it was `svirt_lxc_file_t` which is now `container_file_t`. If that's the case, we can rename this interface to `container_domtrans()` and adjust.

Reply:
Ok, let's change it to `container_domtrans()`, and leave the implementation as is. We can add the other rules in the future if we determine they're needed.